File: doc-examples/mongo-example-java/src/main/java/example/Project.java

package example;

import io.micronaut.data.annotation.EmbeddedId;
import io.micronaut.data.annotation.MappedEntity;

@MappedEntity
public class Project {
    @EmbeddedId
    private ProjectId projectId;
    private String name;

    public Project(ProjectId projectId, String name) {
        this.projectId = projectId;
        this.name = name;
    }

    public ProjectId getProjectId() {
        return projectId;
    }

    public String getName() {
        return name;
    }
}

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/Project.kt
package example

import io.micronaut.data.annotation.EmbeddedId
import io.micronaut.data.annotation.MappedEntity

// tag::project[]
@MappedEntity("projects")
data class Project(@EmbeddedId var projectId: ProjectId, var name: String)
// end::project[]

